src/status_im/ui/screens/desktop/main/add_new/views.cljs

```diff
@@ -2,48 +2,74 @@
-       "Add new contact"]]
-     [react/view {:style {:height 1 :background-color "#e8ebec" :margin-horizontal 16}}]
-     [react/view {:style {:height     90 :margin-horizontal 16 :margin-bottom 16 :background-color :white :border-radius 12}}
+     [react/view {:style {:height 64 :align-items :flex-start :justify-content :center}}
+      [react/text {:style {:font-size 20 :color :black :font-weight "600"}}
+       "New Chat"]]
+     [react/text {:style {:font-size 14}} "Contact code"]
+     [react/view {:style {:height 1 :background-color "#e8ebec"}}]
+     [react/view {:style {:height 90 :margin-bottom 16 :background-color :white :border-radius 12}}
                           ;:box-shadow "0 0.5px 4.5px 0 rgba(0, 0, 0, 0.04)"}}
-      [react/view {:style {:flex-direction :row :margin-horizontal 16 :margin-top 16}}
-       [react/view {:style {:flex 1}}
-        [react/text-input {:placeholder "Contact key"
-                           :flex 1
-                           :on-change   (fn [e]
-                                          (let [native-event (.-nativeEvent e)
-                                                text (.-text native-event)]
-                                            (re-frame/dispatch [:set :contacts/new-identity text])))}]]
-       [react/touchable-highlight {:on-press #(re-frame/dispatch [:add-contact-handler new-contact-identity])}
-        [react/view {:style {:margin-left     16 :width 30 :height 30 :border-radius 15 :background-color "#eef2f5" :align-items :center
+      [react/view {:style {:flex-direction :row :margin-top 16}}
+       [react/text-input {:placeholder "0x..."
+                          :flex 1
+                          :style {:font-size 14 :background-color "#eef2f5" :margin-right 12}
+                          :on-change   (fn [e]
+                                         (let [native-event (.-nativeEvent e)
+                                               text (.-text native-event)]
+                                           (re-frame/dispatch [:set :contacts/new-identity text])))}]
+       [react/touchable-highlight {:on-press #(when-not chat-error (re-frame/dispatch [:add-contact-handler new-contact-identity]))}
+        [react/view {:style {:width 140 :height 45 :border-radius 8 :background-color (if chat-error "#eef2f5" "#4360df") :align-items :center
                              :justify-content :center}}
-         [icons/icon :icons/ok]]]]]
+         [react/text {:style {:font-size 16 :color (if chat-error "#939ba1" :white)}} "Start chat"]]]]]
+     ^{:key "choosecontact"}
+     [react/view
+      (when (seq contacts) [react/text {:style {:font-size 14}} "Or choose a contact"])
+      [react/view {:style {:margin-top 12}} 
+       (doall
+        (for [c contacts]
+        ^{:key (:whisper-identity c)}
+        [react/touchable-highlight {:on-press #(do
+                                                 (re-frame/dispatch [:set :contacts/new-identity (:whisper-identity c)])
```

Comment:
might this PR also fix #4276? these remind me of the updated handlers that Andrea was mentioning.

Reply:
yes that's one of the handlers you can use

Reply:
nice, thanks